server/remote_cache/digest/digest.go

```diff
@@ -174,6 +174,9 @@ func ExtractDigestFromUploadResourceName(resourceName string) (string, *repb.Dig
 }
 
 func ExtractDigestFromDownloadResourceName(resourceName string) (string, *repb.Digest, error) {
+	if strings.Contains(resourceName, "/ac/") {
+		return extractDigest(strings.Replace(resourceName, "ac/", "", 1), downloadRegex)
+	}
 	return extractDigest(resourceName, downloadRegex)
```

Comment:
We don't want to just check for /ac/ because a valid instance name could be /w/ac/k (for example) and that would trigger this condition when it shouldn't.

Instead what you can do is add a method to digest.go like:

ExtractDigestFromActionCacheResourceName(resourceName string) (string, *repb.Digest, error) {
  return extractDigest(resourceName, actionCacheRegex)
}

And then define a regex for your ac stuff like:

actionCacheRegex = regexp.MustCompile("^(?:(?P<instance_name>.*)/)?ac/(?P<hash>[a-f0-9]{64})/(?P<size>\\d+)")

Then you can call ExtractDigestFromActionCacheResourceName from your code in bytestream.go